roxie/ccd/ccdstate.cpp

```diff
@@ -966,69 +968,92 @@ class CRoxieQuerySetManager : public CInterface, implements IRoxieQuerySetManage
                 {
-                    package = packages.queryPackage(id);  // Look for an exact match, then a fuzzy match, using query name as the package id
-                    if(!package) package = packages.matchPackage(id);
-                    if (!package) package = &queryRootRoxiePackage();
+                    // we don't want a single bad query in the set to stop us loading all the others
+                    StringBuffer msg;
+                    msg.appendf("Failed to load query %s from %s", id ? id : "(null)", dllName ? dllName : "(null)");
+                    EXCLOG(E, msg.str());
+                    if (id)
+                    {
+                        StringBuffer emsg;
+                        E->errorMessage(emsg);
+                        Owned<IQueryFactory> dummyQuery = loadQueryFromDll(id, NULL, queryRootRoxiePackage(), NULL, false);
+                        dummyQuery->suspend(emsg.str());
+                        queryHashes[i] = dummyQuery->queryHash();
+                        addQuery(id, dummyQuery.getClear());
+                    }
+                    E->Release();
                 }
-                assertex(package && QUERYINTERFACE(package, const IRoxiePackage));
-                addQuery(id, loadQueryFromDll(id, queryDll.getClear(), *QUERYINTERFACE(package, const IRoxiePackage), &query, forceRetry), hash);
-            }
-            catch (IException *E)
-            {
-                // we don't want a single bad query in the set to stop us loading all the others
-                StringBuffer msg;
-                msg.appendf("Failed to load query %s from %s", id ? id : "(null)", dllName ? dllName : "(null)");
-                EXCLOG(E, msg.str());
-                if (id)
+            });
+            for (auto h : queryHashes)
+                hash = rtlHash64Data(sizeof(h), &h, hash);
+        }
+        unsigned numAliases = const_cast<IPropertyTree *>(querySet)->getCount("Alias");
+        if (numAliases)
+        {
+            std::vector<hash64_t> aliasHashes(numAliases);
+            asyncFor(numAliases, parallelLoadQueries, [this, querySet, &aliasHashes](unsigned i)
+            {
+                aliasHashes[i] = 0;
+                VStringBuffer xpath("Alias[%u]", i+1);
+                IPropertyTree *item = querySet->queryPropTree(xpath);
+                assertex(item);
+                const char *alias = item->queryProp("@name");
+                const char *original = item->queryProp("@id");
+                try
                 {
-                    StringBuffer emsg;
-                    E->errorMessage(emsg);
-                    Owned<IQueryFactory> dummyQuery = loadQueryFromDll(id, NULL, queryRootRoxiePackage(), NULL, false);
-                    dummyQuery->suspend(emsg.str());
-                    addQuery(id, dummyQuery.getClear(), hash);
+                    addAlias(alias, original);
+                    aliasHashes[i] = rtlHash64VStr(alias, aliasHashes[i]);
```

Comment:
minor: clearer to pass 0 as the second parameter.
Arguably clearer to use a temporary variable instead of assigning to aliasHashes[i] and then using on the next line.